src/logic/map_objects/tribes/worker_program.cc

```diff
@@ -341,6 +344,8 @@ void WorkerProgram::parse_findobject(Worker::Action* act, const std::vector<std:
 			act->iparam2 = MapObjectDescr::get_attribute_id(item.second);
 		} else if (item.first == "type") {
 			act->sparam1 = item.second;
+		} else if (item.first == "name") {
+			act->sparam2 = item.second;
```

Comment:
I believe they are registetred at this state, as otherwise we could not determine their attribute id. I think the comment of @Noordfrees was meant to give a clear error message if a wrong name is declared in workerprogram

Reply:
Most probably I was wrong. As I tried to implement a check yesterday and it seems that at least some mapobjects are not loaded yet at this stage. I need to find out if they may be registered so we could implement the check by just trying to load them.